vendor/bat-native-ledger/src/bat/ledger/internal/publisher/server_publisher_fetcher.cc

```diff
@@ -0,0 +1,317 @@
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace {
+
+constexpr size_t kQueryPrefixBytes = 2;
+
+int64_t GetCacheExpiryInSeconds(bat_ledger::LedgerImpl* ledger) {
+  DCHECK(ledger);
+  // NOTE: We are reusing the publisher prefix list refresh interval for
+  // determining the cache lifetime of publisher details. At a later
+  // time we may want to introduce an additional option for this value.
+  return ledger->GetUint64Option(ledger::kOptionPublisherListRefreshInterval);
+}
+
+ledger::PublisherStatus PublisherStatusFromMessage(
+    const publishers_pb::ChannelResponse& response) {
+  switch (response.wallet_connected_state()) {
+    case publishers_pb::UPHOLD_ACCOUNT_KYC:
+      return ledger::PublisherStatus::VERIFIED;
+    case publishers_pb::UPHOLD_ACCOUNT_NO_KYC:
+      return ledger::PublisherStatus::CONNECTED;
+    default:
+      return ledger::PublisherStatus::NOT_VERIFIED;
+  }
+}
+
+ledger::PublisherBannerPtr PublisherBannerFromMessage(
+    const publishers_pb::SiteBannerDetails& banner_details) {
+  auto banner = ledger::PublisherBanner::New();
+
+  banner->title = banner_details.title();
+  banner->description = banner_details.description();
+
+  if (!banner_details.background_url().empty()) {
+    banner->background =
+        "chrome://rewards-image/" + banner_details.background_url();
+  }
+
+  if (!banner_details.logo_url().empty()) {
+    banner->logo = "chrome://rewards-image/" + banner_details.logo_url();
+  }
+
+  for (auto& amount : banner_details.donation_amounts()) {
+    banner->amounts.push_back(amount);
+  }
+
+  if (banner_details.has_social_links()) {
+    auto& links = banner_details.social_links();
+    if (!links.youtube().empty()) {
+      banner->links.insert(std::make_pair("youtube", links.youtube()));
+    }
+    if (!links.twitter().empty()) {
+      banner->links.insert(std::make_pair("twitter", links.twitter()));
+    }
+    if (!links.twitch().empty()) {
+      banner->links.insert(std::make_pair("twitch", links.twitch()));
+    }
+  }
+
+  return banner;
+}
+
+ledger::ServerPublisherInfoPtr ServerPublisherInfoFromMessage(
+    const publishers_pb::ChannelResponseList& message,
+    const std::string& expected_key) {
+  for (auto& entry : message.channel_responses()) {
+    if (entry.channel_identifier() != expected_key) {
+      continue;
+    }
+
+    auto server_info = ledger::ServerPublisherInfo::New();
+    server_info->publisher_key = entry.channel_identifier();
+    server_info->status = PublisherStatusFromMessage(entry);
+    server_info->address = entry.wallet_address();
+    server_info->updated_at =
+        static_cast<uint64_t>(base::Time::Now().ToDoubleT());
+
+    if (entry.has_site_banner_details()) {
+      server_info->banner =
+          PublisherBannerFromMessage(entry.site_banner_details());
+    }
+
+    return server_info;
+  }
+
+  return nullptr;
+}
+
+bool RemovePadding(base::StringPiece* padded_string) {
```

Comment:
This can be removed after we add a `StringPiece` overload for the `RemovePadding` method in `components/brave_private_cdn/` (#5791).